### PR TITLE
Fix/223

### PR DESCRIPTION
**Pull request by sifnoc**
added mutex for `sendTx` method on layer1 context.

### CODE REVIEW

**Review by wanseob — APPROVED** [review]

File: packages/core/src/context/layer1.ts [hunk 1]
```diff
@@ -216,13 +220,31 @@ export class L1Contract extends ZkopruContract {
     account: Account,
     option?: Tx,
   ): Promise<TransactionReceipt | undefined> {
+    const done = await this.aquireSendTxLock(account.address)
```
Comment (JChanceHud) [inline packages/core/src/context/layer1.ts:223]: The `async-lock` package allows you to return a promise. This function can be refactored like this:

```
const result = await this.sendTxLock.acquire(account.address, () => TxUtil.sendTx(
  tx,
  this.address,
  this.web3,
  account,
  options
))
return result
```

**Comment by JChanceHud** [comment]
I'm concerned about adding a mutex lock in the l1 transaction sender because it will affect everything. If we're _only_ trying to fix #223 I'd rather the workers be combined in one. That said, this _may_ fix the `nonce` race condition we've been periodically encountering (mostly in the integration tests).

I think it's okay to merge we should just test it thoroughly.

File: packages/core/src/context/layer1.ts (hunk 1 above)
Comment (sifnoc) [reply packages/core/src/context/layer1.ts:223]: Awesome! much better, thanks.
I was looking for something simpler like this one but did not get it.

**Comment by sifnoc** [comment]
> I'm concerned about adding a mutex lock in the l1 transaction sender because it will affect everything. If we're _only_ trying to fix #223 I'd rather the workers be combined in one. That said, this _may_ fix the `nonce` race condition we've been periodically encountering (mostly in the integration tests).

I agree with that, but I met the nonce error in the wallet in testing. So I decided to add mutex on the `sendTx` instead of fixing workers in coordinator.

> I think it's okay to merge we should just test it thoroughly.

If I met some delays or something else on sending a transaction, I will look in this part first :)

**Pull request merged**